internal/auth/repository/eventsourcing/handler/user_session.go

```diff
@@ -254,158 +221,98 @@ func (u *UserSession) Reduce(event eventstore.Event) (_ *handler.Statement, err
 				}
 			}
 			return u.updateSession(session, event)
-		case user.UserV1PasswordChangedType,
-			user.UserV1MFAOTPRemovedType,
-			user.UserV1ProfileChangedType,
-			user.UserLockedType,
-			user.UserDeactivatedType,
-			user.HumanPasswordChangedType,
-			user.HumanMFAOTPRemovedType,
-			user.HumanProfileChangedType,
-			user.HumanAvatarAddedType,
-			user.HumanAvatarRemovedType,
-			user.UserDomainClaimedType,
-			user.UserUserNameChangedType,
-			user.UserIDPLinkRemovedType,
-			user.UserIDPLinkCascadeRemovedType,
-			user.HumanPasswordlessTokenRemovedType,
-			user.HumanU2FTokenRemovedType:
-			sessions, err := u.view.UserSessionsByUserID(event.Aggregate().ID, event.Aggregate().InstanceID)
-			if err != nil || len(sessions) == 0 {
-				return err
-			}
-			if err = u.appendEventOnSessions(sessions, event); err != nil {
-				return err
-			}
-			if err = u.view.PutUserSessions(sessions); err != nil {
-				return err
-			}
-			return nil
-		case org.OrgDomainPrimarySetEventType:
-			return u.fillLoginNamesOnOrgUsers(event)
-		case user.UserRemovedType:
+		}), nil
+	case user.UserLockedType,
+		user.UserDeactivatedType:
+		return handler.NewUpdateStatement(event,
+			[]handler.Column{
+				handler.NewCol("passwordless_verification", time.Time{}),
+				handler.NewCol("password_verification", time.Time{}),
+				handler.NewCol("second_factor_verification", time.Time{}),
+				handler.NewCol("second_factor_verification_type", domain.MFALevelNotSetUp),
+				handler.NewCol("multi_factor_verification", time.Time{}),
+				handler.NewCol("multi_factor_verification_type", domain.MFALevelNotSetUp),
+				handler.NewCol("external_login_verification", time.Time{}),
+				handler.NewCol("state", domain.UserSessionStateTerminated),
```

Comment:
Why set all the cols to zero value? Yes we terminate the session and that is handled by the state column right?

Theoretically speaking we can have a factor authenticated, after which we terminate the session. The factor was still authenticated at some point in the past, so the projection should still show that.

Or do we do that everywhere where the session is terminated?